src/routers/dashboard/discordAuth.js

```diff
@@ -0,0 +1,51 @@
+const router = require('express').Router()
+const { fetch } = require("undici");
+const { secret, clientID, callbackURL } = require("../../config.json").discord
+
+router.get('/data', async (req, res) => {
+
+    res.json({ message: 'ok', data: req.session.user })
+    //res.redirect(`https://github.com/login/oauth/authorize?client_id=e22bb7f721ea86f96ddd&redirect_uri=https://beta.skyndalex.xyz/auth/github/callback&scope=repo%20user%20admin:org"`)
+})
+
+router.get('/callback', async (req, res) => {
+    if (!req.query.code) return res.send({ message: 'Query code is invalid!' })
+
+    const params = new URLSearchParams()
+
+    params.set('grant_type', 'authorization_code')
+    params.set('code', req.query.code)
+    params.set('redirect_uri', `https://api.skyndalex.xyz/auth/dashboard/discord/callback`)
+
+    const response = await fetch('https://discord.com/api/oauth2/token', {
+        method: 'POST',
+        body: params.toString(),
+        headers: {
+            authorization: `Basic ${btoa(`${clientID}:${secret}`)}`,
```

Comment:
```suggestion
    params.set('grant_type', 'authorization_code')
    params.set('code', req.query.code)
    params.set('redirect_uri', `https://api.skyndalex.xyz/auth/dashboard/discord/callback`)
    params.set('client_secret', secret)

    const response = await fetch('https://discord.com/api/oauth2/token', {
        method: 'POST',
        body: params.toString(),
        headers: {
```
Dokumentacja twierdzi, że secret tutaj powinien być podany w body, a nie w headerze [Dokumentacja discorda](https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-redirect-url-example)